core/modules/savers/put.js

```diff
@@ -0,0 +1,65 @@
+/*\
+title: $:/core/modules/savers/put.js
+type: application/javascript
+module-type: saver
+
+Saves wiki by performing a PUT request to the server
+
+Works with any server which accepts a PUT request
+to the current URL, such as a WebDAV server.
+
+\*/
+(function(){
+
+/*jslint node: true, browser: true */
+/*global $tw: false */
+"use strict";
+
+/*
+Select the appropriate saver module and set it up
+*/
+var PutSaver = function(wiki) {
+	this.wiki = wiki;
+};
+
+PutSaver.prototype.save = function(text,method,callback) {
+	var req = new XMLHttpRequest();
+	req.addEventListener("load", function() {
+		if (this.status === 200 || this.status === 201) {
```

Comment:
I just realised that this error checking needs completion. We need `callback(null);` if there was no error, and `callback(http.responseText);` if there was a problem.